contour/src/main/kotlin/com/squareup/contour/wrappers/ParentGeometry.kt

```diff
@@ -25,15 +26,22 @@ import com.squareup.contour.utils.toYInt
 
 internal class ParentGeometry(
   private val widthConfig: SizeConfig,
-  private val heightConfig: SizeConfig
+  private val heightConfig: SizeConfig,
+  private val paddingProvider: () -> Rect
 ) : Geometry {
-  override fun left(): XInt = XInt.ZERO
-  override fun right(): XInt = widthConfig.resolve().toXInt()
+  override fun left(): XInt = XInt.ZERO + paddingProvider().left
```

Comment:
Would `paddingConfig()` be more consistent with `SizeConfig`?

Reply:
Or just `padding()`, similar to the constraint functions `top()`, `bottom()`, etc.?

Reply:
One of the issues I've experienced while using paddings in my layout blocks is being forced to use `this@ParentView`:

```kotlin
class SomeView : ContourLayout {
  val textView = TextView(context).layoutBy {
    LayoutSpec(
      x = leftTo { parent.top() + this@SomeView.paddingTop },
      y = ...
    )
  }
}
```

If `paddingProvider` is renamed to `padding()` then we could make it public, letting them use `parent.padding().top`. What do you think?

Reply:
Ohh, interesting. I don't see why not! The input `paddingProvider` needs to be a lambda so it updates on the fly when padding has changed, but there's no reason we can't have a public `padding()` getter.

Also agree with `paddingConfig` instead. Making these changes now